dashboard/lib/services/user/pii_scrubber.rb

```diff
@@ -0,0 +1,108 @@
+      end
+
+      def call
+        scrub_user
+        scrub_legacy_data
+        mark_scrubbed
+        @user.save!
+      end
+
+      private def scrub_user
+        # Email and authentication
+        @user.authentication_options.with_deleted.each do |ao|
+          ao.destroy unless ao.destroyed?
+          ao.assign_attributes(email: '', hashed_email: '', authentication_id: nil, data: nil)
+          ao.save(validate: false) # Records may be invalid due to duplicate empty string emails
+        end
+        @user.lti_user_identities.with_deleted.order(deleted_at: :desc).each(&:really_destroy!)
+        @user.email = ''
+        @user.hashed_email = nil
+        @user.encrypted_password = nil
+        @user.reset_password_token = nil
+        @user.unlock_token = nil
+        @user.oauth_refresh_token = nil
+        @user.oauth_token = nil
+        @user.oauth_token_expiration = nil
+        @user.parent_email = nil
+        @user.secret_picture_id = nil
+        @user.secret_words = nil
+        @user.studio_person&.destroy # Studio person record may contain emails
+
+        # Users might have multiple accounts with the same email address.
+        # If there is a live user with the same email, these data points will not be scrubbed.
+        if @email.present? && ::User.find_by_email(@email).blank?
+          MailJet.delete_contact(@email) if @email.present? # Removes the contact from MailJet
+          EmailPreference.where(email: @email).destroy_all
+          census_submissions = Census::CensusSubmission.where(submitter_email_address: email)
+          csfms = Census::CensusSubmissionFormMap.where(census_submission_id: census_submissions.pluck(:id))
+          csfms.destroy_all
+          census_submissions.destroy_all
+        end
+
+        # Names
+        @user.name = nil
+        @user.family_name = nil
+        @user.username = SecureRandom.alphanumeric(20)
+        @user.ops_first_name = nil
+        @user.ops_last_name = nil
+
+        # Address and location data
+        @user.full_address = nil
+        @user.current_sign_in_ip = nil
+        @user.last_sign_in_ip = nil
+        @user.data_transfer_agreement_request_ip = nil
+        @user.user_geos.each(&:clear_user_geo)
+      end
+
+      # Deletes PII from deprecated tables that no longer have a corresponding ActiveRecord model.
+      # Also deletes PII from Pegasus DB, which is planned to for deprecation and should not generally
+      # be accessed from Dashboard code. Pegasus DB is a separate database in the RDS cluster, and includes
+      # PII such as:
+      # - PD applications and forms with email addresses, names, addresses
+      # - Email addresses in Poste tables
+      # - Location and email data in "forms" tables
+      # - Email addresses in contact rollup tables
+      private def scrub_legacy_data
+        if @email.present?
+          @delete_accounts_helper.clean_and_destroy_pd_content(@user.id, @email)
```

Comment:
Please use attr readers
```suggestion
          delete_accounts_helper.clean_and_destroy_pd_content(user.id, email)
```